aws_utils/s3/s3_utils.py

```diff
@@ -40,14 +41,30 @@ def get_filesize(bucket, path):
     return bucket.lookup(path).size
 
 
-def save_to_s3(bucket, path, data):
-    """
-    Takes a data string and saves it to provided path in the provided bucket
+def save_to_s3(bucket, path, data, compress=False):
+    """Takes a data string and saves it to provided path in the provided bucket
+
+    Args:
+        compress (bool): If True the data is gzip compressed before saving to s3.
+        data (str): String of data we wish to pass
+        path (str): Path within the bucket to save the file to, should not contain the bucket name
+        bucket (str or Bucket): Bucket to add the file to, if a string is provided, we try and open an amazon bucket with that name.
     """
+    if isinstance(bucket, str):
```

Comment:
The idea is it can be a bucket or a string, I think it should work at 56 if a bucket, I could add a TypeError if not?